dashboard/src/scenes/stats/charts.js

```diff
@@ -102,39 +117,53 @@ export const CustomResponsiveBar = ({ title, data, categories, onItemClick, axis
               </tr>
             ))}
-            <tr>
-              <td className="tw-border tw-border-zinc-400 tw-p-1 tw-font-bold">Total</td>
-              <td className="tw-border tw-border-zinc-400 tw-p-1 tw-text-center tw-font-bold">{total}</td>
-              <td className="tw-border tw-border-zinc-400 tw-p-1 tw-text-center tw-font-bold">100%</td>
-            </tr>
+            {!isMultiChoice && (
+              <tr>
+                <td className="tw-border tw-border-zinc-400 tw-font-bold">Total</td>
+                <td className="tw-border tw-border-zinc-400 tw-text-center tw-font-bold">{total}</td>
+                {total ? <td className="tw-border tw-border-zinc-400 tw-text-center tw-font-bold">100%</td> : <></>}
+              </tr>
+            )}
           </tbody>
         </table>
       </div>
       <div
         className={[
-          'tw-flex tw-h-80 tw-max-w-[50%] tw-basis-1/2 tw-items-center tw-justify-center tw-font-bold print:tw-w-[600px] print:tw-max-w-[60%] print:!tw-grow print:!tw-basis-0',
+          'tw-relative tw-flex tw-h-80 tw-max-w-[50%] tw-basis-1/2 tw-items-center tw-justify-center tw-font-bold print:tw-w-[600px] print:tw-max-w-[60%] print:!tw-grow print:!tw-basis-0',
           !!onItemClick ? '[&_rect]:tw-cursor-pointer' : '',
         ].join(' ')}>
+        {!!showWarning && (
+          <div className="tw-l-0 tw-r-0 tw-absolute tw-top-0 -tw-mt-5">
+            <p className="tw-m-0 tw-mx-auto tw-w-3/4 tw-text-center tw-text-xs tw-font-normal tw-text-gray-500">
+              On n'affiche sur le graphique que les 15 premières catégories, et les autres seulement si elles représentent plus de 1% des données.
+            </p>
+          </div>
+        )}
         <ResponsiveBar
-          data={data.filter((c) => c.name !== 'Non renseigné')}
-          keys={categories.filter((c) => c !== 'Non renseigné')}
+          data={chartData}
+          keys={categories}
           onClick={onClick}
           indexBy="name"
-          margin={{ top: 40, right: 0, bottom: 50, left: 60 }}
+          margin={{ top: 10, right: 0, bottom: 60, left: 60 }}
           padding={0.3}
+          maxValue={originalDatasetLength}
           valueScale={{ type: 'linear' }}
           indexScale={{ type: 'band', round: true }}
           colors={{ scheme: 'set2' }}
           borderColor={{ from: 'color', modifiers: [['darker', 1.6]] }}
           axisTop={null}
           axisRight={null}
           axisBottom={{
-            tickSize: 5,
+            tickSize: 2,
             tickPadding: 5,
-            tickRotation: 0,
+            tickRotation: -15,
             legend: axisTitleX,
             legendPosition: 'middle',
-            legendOffset: 35,
+            legendOffset: 50,
+            legendTextStyle: {
+              fontSize: 12,
+              fontWeight: 'normal',
+            },
           }}
           axisLeft={{
             tickSize: 5,
```

Comment:
Bonus pour ne pas avoir de chiffres à virgules sur le côté

```suggestion
            tickSize: 5,
            format: (e) => (e ? Math.floor(e) === e && e : ''),
```